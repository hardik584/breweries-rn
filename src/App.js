import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {Button} from 'react-native';
import BreweriesHome from './BreweriesHome';
import BreweriesDetails from './BreweriesDetails';


const Stack = createNativeStackNavigator();

function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="BreweriesHome"
        screenOptions={{
          title: 'My App',
          headerStyle: {backgroundColor: '#38b6ba',    textAlign: 'center',},
          headerTitleStyle: {color: 'white'},
        }}>
          
        <Stack.Screen
          name="BreweriesHome"
          component={BreweriesHome}
          options={{title: 'Breweries Home'}}
        />
        <Stack.Screen
          name="BreweriesDetails"
          component={BreweriesDetails}
          options={{title: 'Breweries Details'}}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default App;
