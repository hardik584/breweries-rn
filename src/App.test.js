import renderer from 'react-test-renderer';
import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render, screen, fireEvent} from '@testing-library/react-native';

import App from './App';

it(`Checking App Rendering`, () => {
  const tree = renderer.create(<App> </App>);
  expect(tree).toMatchSnapshot();
});
