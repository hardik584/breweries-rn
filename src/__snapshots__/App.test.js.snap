// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking App Rendering 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="#38b6ba"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Breweries Home"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "padding": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                }
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Search Here"
                style={
                  Object {
                    "borderColor": "#000000",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flex": 1,
                    "height": 50,
                    "padding": 10,
                  }
                }
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#38b6ba",
                    "borderColor": "black",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "marginLeft": 5,
                    "opacity": 1,
                    "padding": 15,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "https://img.icons8.com/ios-glyphs/344/search.png",
                    }
                  }
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
              </View>
            </View>
            <ActivityIndicator />
          </View>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
